addons/full-import/database.functions

```diff
@@ -0,0 +1,168 @@
+  fi
+
+  db_name=`get_db_name $pf_conf`
+
+  sub_splitter
+  echo "Recreating database $db_name"
+  mysql ${mariadb_args} -e "drop database if exists $db_name; create database $db_name"
+
+  sub_splitter
+  # We reimport the schema so that we have the functions and triggers
+  # Detect minor version of the restored data
+  restoring_version=`egrep -o '[0-9]+\.[0-9]+\.[0-9]+$' usr/local/pf/conf/pf-release | egrep -o '^[0-9]+\.[0-9]+'`
+  echo "Importing bare schema for $restoring_version"
+  mysql ${mariadb_args} $db_name < /usr/local/pf/db/pf-schema-$restoring_version.sql
+  check_code $?
+
+  sub_splitter
+  echo "Replacing create statements from the dump and removing drop statements"
+  # This is done so that tables aren't dropped 
+  sed -i "s/DROP TABLE IF EXISTS /DELETE FROM /g" $dump_file
+  sed -i "s/CREATE TABLE /CREATE TABLE IF NOT EXISTS /g" $dump_file
+
+  sub_splitter
+  echo "Importing $dump_file into $db_name"
+  mysql ${mariadb_args} $db_name < $dump_file
+	check_code $?
+  
+  sub_splitter
+  echo "Importing grants"
+  mysql ${mariadb_args} < $grants_file
+	check_code $?
+
+}
+
+function import_mariabackup() {
+  if ! which qpress > /dev/null 2>&1; then
+    echo "Missing qpress command. Please read the documentation of this script in the Installation guide to prepare your environment before running this script"
+    exit 1
+  fi
+
+  if ! which mariabackup > /dev/null 2>&1; then
+    echo "Missing qpress command. Please read the documentation of this script in the Installation guide to prepare your environment before running this script"
+    exit 1
+  fi
+
+  dump_file="$1"
+
+  sub_splitter
+  echo "Extracting the MariaDB backup"
+  mkdir restore/
+  check_code $?
+  mv $dump_file restore/
+  check_code $?
+  pushd restore/
+  check_code $?
+  mbstream -x < $dump_file
+  check_code $?
+  mv $dump_file ../
+  check_code $?
+
+  sub_splitter
+  echo "Processing .qp files"
+  for i in $(find -name "*.qp"); do 
+    qpress -vd $i $(dirname ${i})
+    check_code $?
+    rm -f $i
+  done
+
+  sub_splitter
+  echo "Applying innodb logs"
+  mariabackup --prepare --target-dir=./
+
+  sub_splitter
+  echo "Stopping packetfence-mariadb and backing up the current data"
+  systemctl stop packetfence-mariadb
+  pkill -9 -f mariadbd
+
+  mv /var/lib/mysql/ "/var/lib/mysql-`date +%s`"
+  mkdir /var/lib/mysql
+  mariabackup --innobackupex --defaults-file=/usr/local/pf/var/conf/mariadb.conf      --move-back --force-non-empty-directories ./
+  chown -R mysql: /var/lib/mysql
+
+  sub_splitter
+  echo "Removing old ib_logfile"
+  rm -f /var/lib/mysql/ib_logfile*
+
+  sub_splitter
+  echo "Starting MariaDB"
+  systemctl start packetfence-mariadb
+  check_code $?
+
+  sub_splitter
+  echo "Performing upgrade of MariaDB. Enter the MariaDB root password if prompted to"
+  echo "NOTE: The root password you must enter is the one of the server of which you're reimporting the data, not this server's MariaDB root password"
+  mysql_upgrade -p
```

Comment:
It looks like `mysql_upgrade` is only done when you did an export on a system which has `mariabackup`. However, standalone installations running 10.3 (without `mariabackup`) need to have their DB upgraded too using this tool.

Reply:
When you restore a mysqldump on the server, it already runs the right version of MariaDB and all the tables are recreated so I don't think there is a need to upgrade the MariaDB files since only raw SQL is reimported, not the actual table files

That is unless I'm missing something else mysql_upgrade is doing

Reply:
Ok, I will check that.